### DISCUSSION TITLE
[QUESTION][iOS] is setBackgroundMessageHandler supported in latest version of messsaging module?

### DISCUSSION THREAD

**Discussion by DovletAmanov** [invertase/react-native-firebase#4485] [Q&A]
I followed the doc and did the setup properly, but I couldn't make **setBackgroundMessageHandler** working. Notification is arriving but event is not emitted. In my case, only **_background notification event_** is **not** being emitted, other events are ok. Is **setBackgroundMessageHandler** supported in iOS? Btw, I read that [here](https://www.reddit.com/r/reactnative/comments/iwc7nl/setbackgroundmessagehandler_not_working_in_ios/)  Apple doesn't give control to the app if it is in background? Any suggestion/help is appreciated, thanks in advance.

"react-native": "0.63.3",
"@react-native-firebase/app": "^8.4.7",
"@react-native-firebase/messaging": "^7.9.2",

**Comment by andersonaddo** [comment]
It's supported and working. 
Provide more info:
How do you know the notification is arriving?
What does your notification look like (structure wise)
Sample code

**Reply by DovletAmanov** [reply]
1 - _How do you know the notification is arriving?_ -> Notification bar is appearing on device when notification is sent from FCM notification composer

2 - _What does your notification look like (structure wise)_ -> I use FCM + react-native-push-notifications + @react-native-community/push-notification-ios and I call these listener methods below in **index.js** before calling main App component:

```
PushNotification.configure({ ...configs as in doc... });

messaging().setBackgroundMessageHandler(async (remoteMessage) => {
  console.log('Message handled in the background!', remoteMessage);
});

messaging().onNotificationOpenedApp((remoteMessage) => {
  console.log('FIREBASE IOS Background', remoteMessage);
});

messaging().onMessage(async (remoteMessage) => {
  console.log('A new FCM message arrived!', JSON.stringify(remoteMessage));
});
```
